# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/forms/nsTextControlFrame.h
# Commit: 928b78db7b30
# Full Hash: 928b78db7b30e53ea59564c2001a411245a7d3fb
# Author: David Shin <dshin@mozilla.com>
# Date: 2023-08-12 09:15:25
# Regressor Bug: 1846982
# File Overlap Count: 1
# Description:
#   Bug 1846982: Baseline calculation for textarea. r=emilio
#   
#   Tests are added as tentative, because the behaviour when there's no content,
#   or when there's placeholder text, seem very differnt for each browser.
#   
# ==============================================================================

diff -r a42535df0242 -r 928b78db7b30 layout/forms/nsTextControlFrame.h
--- a/layout/forms/nsTextControlFrame.h	Fri Aug 11 19:14:51 2023 +0000
+++ b/layout/forms/nsTextControlFrame.h	Fri Aug 11 19:24:32 2023 +0000
@@ -79,14 +79,7 @@
 
   Maybe<nscoord> GetNaturalBaselineBOffset(
       mozilla::WritingMode aWM, BaselineSharingGroup aBaselineGroup,
-      BaselineExportContext) const override {
-    if (!IsSingleLineTextControl()) {
-      return Nothing{};
-    }
-    NS_ASSERTION(!IsSubtreeDirty(), "frame must not be dirty");
-    return GetSingleLineTextControlBaseline(this, mFirstBaseline, aWM,
-                                            aBaselineGroup);
-  }
+      BaselineExportContext aExportContext) const override;
 
   static Maybe<nscoord> GetSingleLineTextControlBaseline(
       const nsIFrame* aFrame, nscoord aFirstBaseline, mozilla::WritingMode aWM,